### PR TITLE
[HttpFoundation] Deprecate Request::getSession() when Request::hasSession() is false

### LINKED ISSUE #9457: [HttpFoundation] deprecate call to Request::getSession() when Request::hasSession() returns false
Before some command returned `null` but now it explodes the application with an exception. See https://github.com/symfony/symfony/pull/26564

### PR DESCRIPTION
**Pull request by snoek09**
This fixes #9457.

### CODE REVIEW

File: components/http_foundation.rst [hunk 1]
```diff
@@ -235,6 +235,13 @@ the
 method tells you if the request contains a session which was started in one of
 the previous requests.
 
+.. versionadded:: 4.1
+
+    Using :method:`Symfony\\Component\\HttpFoundation\\Request::getSession()` when no session
+    has been set, was deprecated in Symfony 4.1. It will throw an exception in
+    Symfony 5.0 when session is null.
+    Use :method:`Symfony\\Component\\HttpFoundation\\Request::hasSession()` instead.
```
Comment (xabbuh) [inline components/http_foundation.rst]: I would this reword to say something like "Check for an existing session first by calling `Request::hasSession()`."

**Comment by xabbuh** [comment]
@snoek09 I would still update `form_login.rst` to check if the session actually exists.

**Review by xabbuh — APPROVED** [review]

File: components/http_foundation.rst [hunk 2]
```diff
@@ -235,6 +235,12 @@ the
 method tells you if the request contains a session which was started in one of
 the previous requests.
 
+.. versionadded:: 4.1
+    Using :method:`Symfony\\Component\\HttpFoundation\\Request::getSession()`
+    when no session has been set, was deprecated in Symfony 4.1. It will throw
```
Comment (xabbuh) [inline components/http_foundation.rst]: the comma could be removed

File: components/http_foundation.rst [hunk 3]
```diff
@@ -235,6 +235,12 @@ the
...
+    an exception in Symfony 5.0 when session is null. Check for an existing session
```
Comment (xabbuh) [inline components/http_foundation.rst]: [...] when the session [...]

Comment (xabbuh) [reply components/http_foundation.rst]: and "null" should be enclosed by double backticks

**Comment by javiereguiluz** [comment]
A nice deprecation message! Thanks for contributing it.

**Pull request closed without merging**